dev: spruce up help text

irfansharif opened the pull request:
Just cribbed from `make help` + a partial revert of cockroachdb#74571.

Release note: None

irfansharif committed:
dev: spruce up help text

Just cribbed from `make help`.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/74587)
<!-- Reviewable:end -->

irfansharif committed:
dev: skip shim script under bin/

Didn't intend to, pushed out a wrong revision as part of #74571.

Release note: None

irfansharif commented on dev:
```diff
@@ -14,17 +14,6 @@ if [ ! -f "$BINARY_PATH" ]; then
-#!/usr/bin/env bash
-
-# Code generated by top-level 'dev' script; DO NOT EDIT.
-${BINARY_PATH}
```
This was missing a `"\$@"` 😔

itsbilal reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 6 of 6 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74587#-:-MsqtFxE8qhd1shqgDiL:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rickystewart)



<!-- Sent from Reviewable.io -->

irfansharif commented:
bors r+
